src/main/ruby/core/dir.rb

```diff
@@ -49,22 +53,53 @@ def initialize(path, options=undefined)
       enc = Rubinius::Type.coerce_to_encoding enc if enc
     end
 
-    Truffle.invoke_primitive :dir_open, self, path, enc
+    @encoding = enc || Encoding.find('filesystem')
```

Comment:
Did we ever end up adding our own `Encoding::FILESYSTEM`?

Reply:
No, but I was hesitating to do it in this PR, should I?

Reply:
Saving it for later is fine by me.

Reply:
Done.